### PR TITLE
Add a new property, display_as_unplugged, to External levels

### PR DESCRIPTION
**Pull request by Erin007**
Working on: [LP-278 ](https://codedotorg.atlassian.net/browse/LP-278)

Levelbuilders would like to enable markdown / external html level types to be shown as "unplugged" levels if a check mark is checked accordingly in Levelbuilder. This PR adds that checkbox to the level creation page for external levels and uses its value to set a new property, `display_as_unplugged`.  

Screen Shot 2019-05-01 at 11 29 34 AM

If unchecked: 
unchecked-display-as-unplugged

If checked: 
checked-display-as-unplugged

Up next: this new property will be used to make the level bubble appear the same as an "unplugged" level.

### CODE REVIEW

**Review by dmcavoy — APPROVED** [review]
LGTM

File: dashboard/app/models/levels/external.rb [hunk 1]
```diff
@@ -52,6 +55,10 @@ def concept_level?
     true
   end
 
+  def display_as_unplugged
+    properties.merge({'display_as_unplugged' => !properties['display_as_unplugged']})
```
Comment (islemaster) [inline dashboard/app/models/levels/external.rb]: I'm confused by this.  Why does this method return a hash? And why does it invert the `display_as_unplugged` value?

**Comment by davidsbailey** [comment]
Please also test by adding `display_as_unplugged true` or `display_as_unplugged false` to the .external file and then running `rake seed:dsls`. this should update what you see in the DB.

Please also test that when you load the level again in levelbuilder, that what you see matches what's in the DB.

File: dashboard/app/models/levels/external.rb (hunk 1 above)
Comment (Erin007) [reply dashboard/app/models/levels/external.rb]: oh good call.  The form field is doing more for me than I thought. removed!

File: dashboard/app/models/levels/external.rb [hunk 2]
```diff
@@ -24,6 +24,9 @@
 #
 
 class External < DSLDefined
+  serialized_attrs %w(
+    display_as_unplugged
+  )
```
Comment (davidsbailey) [inline dashboard/app/models/levels/external.rb:29]: I realize you may only part-way through responding to feedback, but I don't think this is the right approach. when you save an external level in levelbuilder, it goes down through the call stack like this:
* https://github.com/code-dot-org/code-dot-org/blob/41d8adc6a86afb0ff08b11f171fa881490736b72/dashboard/app/models/levels/dsl_defined.rb#L156-L158
* https://github.com/code-dot-org/code-dot-org/blob/41d8adc6a86afb0ff08b11f171fa881490736b72/dashboard/app/models/levels/dsl_defined.rb#L70-L74
* https://github.com/code-dot-org/code-dot-org/blob/f749e1c7cbb878c4af832879ed899a691549499e/dashboard/app/dsl/base_dsl.rb#L31-L34

from there, instance eval reads the individual commands in the dsl file (e.g. `display_as_unplugged true`) and calls the corresponding function on the DSL class. So, I believe what you need to do here is remove the serialized_attrs and instead implement a method like the one I posted earlier for teacher_markdown.

Comment (Erin007) [reply dashboard/app/models/levels/external.rb:29]: thanks for the info, Dave!

**Comment by Erin007** [comment]
Closing in favor of https://github.com/code-dot-org/code-dot-org/pull/28325

**Pull request closed without merging**